build-system/tasks/lint.js

```diff
@@ -132,8 +148,30 @@ function lint() {
   if (argv.fix) {
     options.fix = true;
   }
-  if (argv.files) {
-    config.lintGlobs[config.lintGlobs.indexOf('**/*.js')] = argv.files;
+  if (argv.files ||
+      process.env.TRAVIS_PULL_REQUEST ||
+      process.env.LOCAL_PR_CHECK) {
+    if (argv.files) {
+      config.lintGlobs =
+          config.lintGlobs.filter(e => e !== '**/*.js').concat(argv.files);
+    } else {
+      config.lintGlobs =
+          config.lintGlobs.filter(e => e !== '**/*.js').concat(getLintFiles());
+    }
+    // Override .eslintrc settings here.
+    options['rules'] = {
+      // TODO(rsimha, #15255): Make this error by default in .eslintrc.
+      'valid-jsdoc': [2, {
```

Comment:
We could instead maintain this as a second `.eslintrc` (let's say, `.eslintrc-strict`) with an `extends` option pointing to the default `.eslintrc. Then, switch to using the strict version for PRs.

This moves this configuration out of JavaScript code.

Reply:
Done in https://github.com/ampproject/amphtml/pull/15256/commits/bb35a33697a17c9bd55695442d66e3daeefc05c8.